src/main/java/com/odysseusinc/athena/service/mail/EmailServiceImpl.java

```diff
@@ -128,19 +116,32 @@ public void sendVocabulariesWereSharedNotification(AthenaUser recipient, AthenaU
         final String bundleUrl = urlBuilder.downloadVocabulariesLink(bundle.getUuid());
         final Map<String, Object> emailParameters = getParameters(recipient, bundleOwner, bundleUrl, bundle.getCdmVersion(), bundle.getReleaseVersion());
 
-        try {
-            emailSenderService.send(EmailType.VOCABULARIES_SHARED_DOWNLOAD, emailParameters, recipient.getEmail());
-            LOGGER.info("Email with link for download zip is sent to user with id: [{}], zip link: [{}]",
-                    recipient.getId(), bundleUrl);
-        } catch (Exception ex) {
-            emailSenderService.send(EmailType.FAILED_SENDING_TO_ADMIN, getParameters(ex), getAdminEmails());
+        send(EmailType.VOCABULARIES_SHARED_DOWNLOAD, emailParameters, asList(recipient.getEmail()), emptyList(), getAdminEmails());
+        log.info("Email with link for download zip is sent to user with id: [{}], zip link: [{}]", recipient.getId(), bundleUrl);
+    }
+
+    private void send(EmailType messageType, Map<String, Object> parameters, List<String> to, List<String> bcc, List<String> notifyOnFailureEmails) {
+
+        log.debug("Sending {} to {}", messageType, to);
+        final String emailBody = contentBuilder.build(messageType.getTemplate(), parameters);
+        emailSenderService.sendAsync(messageType.getSubject(), emailBody, to, bcc)
+                .exceptionally(ex -> handleError(ex, messageType, emailBody, to, notifyOnFailureEmails));
+    }
+
+    private Void handleError(Throwable ex, EmailType messageType, String emailBody, List<String> to, List<String> notifyOnFailureEmails) {
+
+        log.error("Failed email {} \n\n\n{}\n\n\n, to: {}", messageType, emailBody, to, ex);
+        if (CollectionUtils.isNotEmpty(notifyOnFailureEmails)) {
+            final Map<String, Object> parameters = ImmutableMap.of("exception", ex.getCause());
+            send(EmailType.FAILED_SENDING_TO_ADMIN, parameters, emptyList(), notifyOnFailureEmails, emptyList());
```

Comment:
If I understand correctly it can be infinite recursion in case of network exception(or any other constantly reproducible exception). I would just log the exception for `EmailType.FAILED_SENDING_TO_ADMIN`